app/controllers/votes_controller.rb

```diff
@@ -1,2 +1,32 @@
 class VotesController < ApplicationController
+  before_action :authenticate_user!
+  
+  helper_method :vote
+
+  def create
+    vote = current_user.votes.build(vote_params)
+    vote.save
+    redirect_back(fallback_location: root_path)
+  end
+
+  def update
+    vote.value == 1 ? vote.value = -1 : vote.value = 1
```

Comment:
I'm curious why the toggle approach here over passing a param to set the `value`?

Reply:
I guess its one less thing to pass in locals to the `shared/vote/edit` partial, but then I have a conditional here, so hmm... 🤔